modules/apps/asset/asset-taglib/test/asset_categories_navigation/AssetCategoriesNavigationTreeView.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import '@testing-library/jest-dom';
+
+import '@testing-library/jest-dom/extend-expect';
+import {fireEvent, render, screen} from '@testing-library/react';
+import React from 'react';
+
+import AssetCategoriesNavigationTreeView from '../../src/main/resources/META-INF/resources/js/asset_categories_navigation/AssetCategoriesNavigationTreeView';
+
+const DEFAULT_PROPS = {
+	selectedCategoryId: '',
+	vocabularies: [
+		{
+			children: [
+				{
+					categoryId: 42348,
+					children: [],
+					icon: 'categories',
+					id: '42348',
+					name: 'category',
+					url: 'http://localhost:8080/home/-/categories/42348?p_r_p_categoryId=42348',
+					vocabularyId: '42347',
+				},
+			],
+			disabled: true,
+			icon: 'vocabulary',
+			id: '42347',
+			name: 'vocabulary',
+		},
+	],
+};
+
+describe('AssetCategoriesNavigationTreeView', () => {
+	beforeEach(() => {
+		jest.clearAllMocks();
+	});
+
+	it('tree is expanded when clicked on the vocabulary name', () => {
+		const vocabularyName = DEFAULT_PROPS.vocabularies[0].name;
+		const categoryName = DEFAULT_PROPS.vocabularies[0].children[0].name;
+
+		render(<AssetCategoriesNavigationTreeView {...DEFAULT_PROPS} />);
+
+		expect(screen.getByText(vocabularyName)).toBeInTheDocument();
+		expect(screen.queryByText(categoryName)).not.toBeInTheDocument();
+
+		fireEvent.click(screen.getByText(vocabularyName));
```

Comment:
we could save one query in the DOM if we save `screen.getByText(vocabularyName)` in a variable.